pkg/kubectl/cmd/apply_test.go

```diff
@@ -113,18 +125,17 @@ func readServiceFromFile(t *testing.T, filename string) *api.Service {
 	return &svc
 }
 
-func annotateRuntimeObject(t *testing.T, originalObj, currentObj runtime.Object, kind string) (string, []byte) {
+func annotateRuntimeObject(t *testing.T, originalObj, currentObj runtime.Object,
+	codec runtime.Codec) (string, []byte) {
 	originalAccessor, err := meta.Accessor(originalObj)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	originalLabels := originalAccessor.GetLabels()
-	originalLabels["DELETE_ME"] = "DELETE_ME"
-	originalAccessor.SetLabels(originalLabels)
-	original, err := runtime.Encode(testapi.Default.Codec(), originalObj)
-	if err != nil {
-		t.Fatal(err)
+	if originalLabels != nil {
```

Comment:
@liggitt 
Are you sure adding the DELETE_ME label no matter what is the right thing? 
1) The test fails if I make the modification you have suggested.  (The PATCH request differs from the empty request) 
```
[vagrant@k8s-env cmd]$ go test -v -run TestApplyIdempotentTPREntry
=== RUN   TestApplyIdempotentTPREntry
--- FAIL: TestApplyIdempotentTPREntry (0.00s)
	apply_test.go:614: Unexpected Patch. Found unexpected annotation: {"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"unit-test.test.com/v1\",\"key\":\"value\",\"kind\":\"Widget\",\"metadata\":{\"annotations\":{},\"name\":\"widget\",\"namespace\":\"test\"}}\n"},"labels":null}}
FAIL
```

2) The GET request's response will differ from the actual kubectl execution. 
I execute the following command on a running api server 
` kubectl.sh --kubeconfig=/var/run/kubernetes/admin.kubeconfig  apply -f widget.yaml -v=10`
And the logs show the following info about the GET request 
```
I0211 02:57:19.899574   22691 round_trippers.go:417] GET https://localhost:6443/apis/unit-test.test.com/v1/namespaces/test/widgets/widget 200 OK in 1 milliseconds
I0211 02:57:19.899585   22691 round_trippers.go:423] Response Headers:
I0211 02:57:19.899590   22691 round_trippers.go:426]     Date: Sat, 11 Feb 2017 02:57:19 GMT
I0211 02:57:19.899593   22691 round_trippers.go:426]     Content-Type: application/json
I0211 02:57:19.899596   22691 round_trippers.go:426]     Content-Length: 532
I0211 02:57:19.899614   22691 request.go:991] Response Body: {"apiVersion":"unit-test.test.com/v1","key":"value","kind":"Widget","metadata":{"name":"widget","namespace":"test","selfLink":"/apis/unit-test.test.com/v1/namespaces/test/widgets/widget","uid":"3e03a56e-f005-11e6-8716-0800275ae9e7","resourceVersion":"14220","creationTimestamp":"2017-02-11T02:53:18Z","annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"unit-test.test.com/v1\",\"key\":\"value\",\"kind\":\"Widget\",\"metadata\":{\"annotations\":{},\"name\":\"widget\",\"namespace\":\"test\"}}\n"}}}
```

As you can see the response of the GET request does not have the DELETE_ME label. Is that wise a different behavior to the unit test ?